numpy/tests/test_warnings.py

```diff
@@ -34,9 +34,9 @@ def visit_Call(self, node):
         ast.NodeVisitor.generic_visit(self, node)
 
         if p.ls[-1] == 'simplefilter' or p.ls[-1] == 'filterwarnings':
-            if node.args[0].value == "ignore":
+            if node.args[0].value == "ignore" and 'test' not in self.__filename.name:
                 raise AssertionError(
-                    "warnings should have an appropriate stacklevel; "
+                    "ignore filters should only be used in tests; "
```

Comment:
I'm not sure why this check was put here back in 4ec016784b02f69347c4429829b9eb8225ace5ee (the warning message was subsequently (incorrectly?) screwed up in 6737f241454688f6ea59a883f6c6e3fe8eff8e8b. @seberg do you have context for this check? Are you ok with me adding an exception for tests, given the context of getting rid of `suppress_warnings`?

Reply:
Because they would have gotten added to tests causing pain otherwise. Keeping it around is just that one shoukd avoid them in code too. I am fine either way. But we probably manage to not use them when errstate will do, so this check is also OK to delete.

Reply:
So do what you like here: I am OK with keeping it because we still don't like ignores in code, but I am also happy if you just delete it.

The main reasons for it, don't exist anymore, there is no Chesterton's fence to worry about :).

Reply:
Let's leave the check, IMO it's a good idea for a test to fail if someone adds a warning filter somewhere in the NumPy implementation. I'll see if there's a more fine-grained way to check this using pathlib...